If there is changes, asks user to confirm "load", "new" and leaving page

vsariola opened the pull request:
Stores a deep copy of the song when it is created/loaded/last saved. When the user tries to create a new song, load or leave the page, does a deep comparison between the current song and the song before modifications. If they are not the same, asks the user to confirm before overwriting the song.

For the record, I first tried the solution to track when the song has changed. It proved quite bug-prone, as the song is changed in so many places, and setting the flag was littered throughout the code... I finally settled for this solution.

deepCompare and deepCopy are helper functions that could be replaced with more modern stuff e.g. many general purpose libraries and ES6 come with them, if it feels too ugly to have them in the gui.js.

mbitsnbites commented on gui.js:
```diff
@@ -1931,6 +1944,57 @@ var CGUI = function()
   };
 
+  // based on https://j11y.io/javascript/deep-copying-of-objects-and-arrays/
+  var deepCopy = function(obj) {
```
deepCopy and deepCompare should be moved to common.js

mbitsnbites commented:
Really sweet! Could you please rebase the branch so that it can be merged?

Commit:
Ask user to confirm "new", "load" and leaving page if there's unsaved changes

Stores a deep copy of the song in unmodified state, and does a deep equality comparison
between the unmodified and current song to see if there's need to confirm

vsariola commented:
Rebased. Renamed also "deepCompare" to "deepEquals" (more apt name) and put them to common.js.
